### PR TITLE
Moving more types from reth crate to node core crate

### PR DESCRIPTION
**Pull request by i-m-aditya**
Ref #6063

### CODE REVIEW

**Comment by i-m-aditya** [comment]
@mattsse Have moves args, and other modules on which it is dependent to the node-core crate. Any other refactoring required?

**Comment by i-m-aditya** [comment]
Will fix the failed checks by EOD. But is this the desired approach.

**Review by mattsse — CHANGES_REQUESTED** [review]
this is a good draft!

we don't want that examples need to also add the node-core crate

instead we want to re-export node-core from the bin/reth crate

**Review by mattsse — CHANGES_REQUESTED** [review]
sweet,

last request to also reexport the types from previous locations, so that this change becomes non breaking. Because this is used by a few users already and we don't want to introduce breaking changes incrementally

File: examples/additional-rpc-namespace-in-cli/src/main.rs
Comment (i-m-aditya) [reply examples/additional-rpc-namespace-in-cli/src/main.rs:19]: That would mean having the same modules(args, cli) in both crates: node-core and bin/reth. Should I go ahead and do this change?

Comment (mattsse) [reply examples/additional-rpc-namespace-in-cli/src/main.rs:19]: yes please 🙏

adding a few mods with reexports is fine, we want to reduce breaking changes for now

File: examples/trace-transaction-cli/src/main.rs
Comment (mattsse) [inline examples/trace-transaction-cli/src/main.rs:28]:
```suggestion
```

File: examples/beacon-api-sse/src/main.rs
Comment (mattsse) [inline examples/beacon-api-sse/src/main.rs:23]:
```suggestion
```

File: examples/additional-rpc-namespace-in-cli/src/main.rs
Comment (mattsse) [inline examples/additional-rpc-namespace-in-cli/src/main.rs:23]:
```suggestion
```

File: bin/reth/src/cli/ext.rs
Comment (Rjected) [inline bin/reth/src/cli/ext.rs:211]: should probably keep them relative to a crate so our doc tests can pick up on any missing links

File: bin/reth/src/commands/init_cmd.rs
Comment (Rjected) [inline bin/reth/src/commands/init_cmd.rs:3]: nit: this can probably be moved back to the `crate::{` import section

**Review by mattsse — APPROVED** [review]
lgtm, pending @Rjected 

I assume we need a few smaller followups but since this touches a lot we should merge asap to reduce friction

**Review by Rjected — APPROVED** [review]
have some small nits, otherwise LGTM. for followup: removing any code that's now duplicated in `reth_node_core`

**Pull request merged**
